src/components/QueryEditor/SampleQueries/SampleQueriesModal.tsx

```diff
@@ -0,0 +1,81 @@
+import { css } from '@emotion/css';
+
+import { GrafanaTheme2 } from '@grafana/data';
+import { Modal, useStyles2, Text } from '@grafana/ui';
+
+import React from 'react';
+import Prism from 'prismjs';
+import { flattenTokens, pplTokenizer } from './utils';
+import { sampleQueries } from './sampleQueries';
+import { QueryType } from 'types';
+import { trackSampleQueryClicked } from 'tracking';
+
+type Props = {
+  queryLanguage: QueryType;
+  isOpen: boolean;
+  onClose: () => void;
+  onSelectSampleQuery: (query: string) => void;
+};
+
+export const SampleQueriesModal = (props: Props) => {
+  const styles = useStyles2(getStyles);
+
+  const onSampleSelect = (query: string) => {
+    props.onSelectSampleQuery(query);
+    trackSampleQueryClicked(props.queryLanguage);
+    props.onClose();
+  };
+
+  return (
+    <Modal isOpen={props.isOpen} title="Sample queries" onDismiss={props.onClose}>
+      <Text>Get started by selecting a sample query. You can then edit it in the query field.</Text>
+      {sampleQueries.map((item, i) => {
+        return (
+          <div key={i}>
+            <Text variant="h6" weight="bold">
+              {item.title}
+            </Text>
+            <button
+              type="button"
+              className={styles.cheatSheetExample}
+              key={item.queryString}
+              onClick={() => onSampleSelect(item.queryString)}
+            >
+              <pre data-testid={`sample-query-${i}`}>{renderHighlightedMarkup(item.queryString, `item-${i}`)}</pre>
```

Comment:
```suggestion
              <pre data-testid={`sample-query-${i}`} className={styles.cheatSheetExampleCode}>
                {renderHighlightedMarkup(item.queryString, `item-${i}`)}
              </pre>
```

nit: the clickable area expands below the apparent borders of the button because the `pre` has a default bottom margin. the below suggested styles zero out the margins and add it to the button instead.

before
---
<img width="761" height="506" alt="previous" src="https://github.com/user-attachments/assets/65e60340-fc93-4598-9e45-2833b1e52394" />


after
---
<img width="761" height="506" alt="new" src="https://github.com/user-attachments/assets/a67b6e8f-c736-465d-aca1-a73b7e5c4f24" />